src/handle-data.js

```diff
@@ -31,11 +39,28 @@ export function getColumnInfo(qHyperCube, colIndex) {
       stylingInfo: info.qAttrExprInfo.map((expr) => expr.id),
       sortDirection: directionMap[info.qSortIndicator],
       dataColIdx: colIndex,
+      totalInfo,
     }
   );
 }
 
-export default async function manageData(model, layout, pageInfo, setPageInfo) {
+export function showTotals(layout) {
+  const hasOnlyMeasure = () => layout.qHyperCube.qDimensionInfo.length === 0;
+  const hasTotalsMeasures = () =>
+    layout.qHyperCube.qMeasureInfo.length > 0 && layout.qHyperCube.qGrandTotalRow.length > 0;
+  if (hasOnlyMeasure() || !hasTotalsMeasures()) return false;
+  return true;
+}
+
+export function getTotalPosition(layout) {
+  if (!showTotals(layout)) return 'noTotals';
+  const isTotalModeAuto = () => layout.totals?.show === true;
```

Comment:
this shouldn't be a function

Reply:
also a comment about that totals.true is actually the auto setting, this confused me a lot